### PR TITLE
Fe connect be list of company

### PR DESCRIPTION
**Pull request by asdelay**
## Describe your changes

- I did an awesome feature.
- Refactored code
- Added i18n to search component
- Added cursor: pointer to search icon
- Added debounce function to utils
- Created redux slice for handling companies with RTK Query
- Added white background to Popup Close component
- Added error and loading handler to SuperAdmin component
- Moved SuperAdmin component logic to custom hook

## Issue ticket code (and/or) and link

- [DREAMTEAM-23](https://zenbitdreamteam.atlassian.net/browse/DREAMTEAM-23)

### **General**

- [x] Types for input and output params
- [x] No `any`, should be also added to eslint rules
- [x] try/catch for any async function
- [x] no magic numbers (describe it in constants)
- [x] compare only with constants not with strings (instead of user === ‘adminʼ, better user === roles.admin)
- [x] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no console.log in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL should be in constant file
- [x] constants and function names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyyˮ), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _ʼ, const EACH_DAY_15h_15min = ‘15 3 _ \* \*ʼ

### Frontend

- [x] split component and logic (move logic to custom hook file)
- [x] colors, font size, and font name should be in the theme or in the constants
- [x] no text in the components, use i18n approach, even if you have only one language for now
- [x] inline styles prohibited
- [x] import should be absolute. instead of ../../../components/myComponent should be components/myComponent

### CODE REVIEW

File: src/pages/SuperAdminPage/index.tsx [hunk 1]
```diff
@@ -67,34 +74,51 @@ export const SuperAdminPage = (): React.ReactElement => {
...
-                key={company.id}
-                item={company}
-                setModal={() => setEditModal(true)}
-              />
-            ))}
+            {isLoading ? (
+              <h3>{t('loading')}</h3>
+            ) : (
+              data &&
+              data.data.map((company) => (
```
Comment (lyudad) [inline src/pages/SuperAdminPage/index.tsx]: data && data.data.map => data?.data.map

File: src/pages/SuperAdminPage/index.tsx [hunk 2]
```diff
@@ -67,34 +74,51 @@ export const SuperAdminPage = (): React.ReactElement => {
...
         type={CompanyFormTypes.Add}
       />
       <CompanyForm
         modalOpen={editModalOpen}
-        closeModal={() => setEditModal(false)}
+        closeModal={() => setEditModalOpen(false)}
         type={CompanyFormTypes.Edit}
       />
+      {error && (
+        <SignInPopup type='error' text='Failed to load companies data' />
```
Comment (lyudad) [inline src/pages/SuperAdminPage/index.tsx]: 1.3.3

File: src/utils/useDebounce.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react';
+
+interface UseDebounceProps {
+  searchValue: string;
+  delay: number;
+}
+export const useDebounce = ({
```
Comment (lyudad) [inline src/utils/useDebounce.ts:7]: огонь, это очень круто что не забыл

File: src/components/CompanyPagination/index.tsx
Comment (lyudad) [inline src/components/CompanyPagination/index.tsx:1]: if this component belong to company only, move it to /pages/[name of component]/components/CompanyPagination

File: src/pages/SuperAdminPage/index.tsx (hunk 2 above)
Comment (asdelay) [reply src/pages/SuperAdminPage/index.tsx]: I'm so sorry. Checked through multiple times before PR but didn't see(. Fixed

File: src/pages/SuperAdminPage/index.tsx (hunk 1 above)
Comment (asdelay) [reply src/pages/SuperAdminPage/index.tsx]: Fixed

File: src/components/CompanyPagination/index.tsx
Comment (asdelay) [reply src/components/CompanyPagination/index.tsx:1]: Fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**